### PR TITLE
fix(API): Create Custom Metadata endpoint fix [TSI-2222]

### PR DESCRIPTION
**Pull request by jablan**
Update Custom Metadata non-GET endpoints so that they comply with our standard way of passing parameters.

### CODE REVIEW

File: clients/go/test/api_custom_metadata_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	t.Run("Test CustomMetadataService CustomMetadataCreate", func(t *testing.T) {
+		customMetadataPropertiesCreateParameters := phrase.CustomMetadataPropertiesCreateParameters{
+			Name:        "my_property",
+			DataType:    phrase.STRING,
+			Description: "my description",
+			ProjectIds:  []string{"project_id", "project_id2"},
+		}
+		localVarOptionals := phrase.CustomMetadataPropertyCreateOpts{}
+		resp, httpRes, err := apiClient.CustomMetadataApi.CustomMetadataPropertyCreate(context.Background(), "account_id", customMetadataPropertiesCreateParameters, &localVarOptionals)
```
Comment (hahmed-dev) [inline clients/go/test/api_custom_metadata_test.go:46]: Just one question passing `customMetadataPropertiesCreateParameters` here is equivalent to passing these arguments individually 

```
(ctx _context.Context, accountId string, name string, dataType CustomMetadataDataType
```

in the actual api `/clients/go/api_custom_metadata.go`?

Comment (jablan) [reply clients/go/test/api_custom_metadata_test.go:46]: no, these individual arguments are replaced by this PR with `CustomMetadataPropertiesCreateParameters` struct. technically, it's a breaking change, but I don't think anyone is using this yet.

File: paths/custom_metadata_properties/update.yaml [hunk 2]
```diff
@@ -8,40 +8,40 @@ parameters:
...
-  in: query
-  schema:
-    type: array
-    items:
-      type: string
+requestBody:
+  required: true
+  content:
+    application/json:
+      schema:
```
Comment (hahmed-dev) [inline paths/custom_metadata_properties/update.yaml:15]: Should we also add the required `name` and `datatype` section?
```suggestion
      schema:
        required:
        - name
        - data_type
```

Comment (jablan) [reply paths/custom_metadata_properties/update.yaml:15]: yeah, I haven't checked the backend code, but I would guess that the data type is not changeable once the property is created, and the name is certainly not required, if one doesn't want to change it.

**Review by hahmed-dev — APPROVED** [review]

**Pull request merged**